### PR TITLE
Add capability to collapse all grouping in topology view

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**Fixes**: 
https://issues.redhat.com/browse/ODC-4316

**Description**: 
As a developer, I want to view all my “Grouped Resources” (Application Groups, Helm Releases, Knative Services, Operator Backed Services)as collapsed groupings in the Topology Graph View

**Screen shots / Gifs for design review**:

image

image

**Browser conformance**: 
- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

cc @openshift/team-devconsole-ux @serenamarie125 @beaumorley @bgliwa01

/kind feature

### CODE REVIEW

**Comment by christianvogt** [comment]
@beaumorley is there a design for this feature somewhere?

I don't like how the filters become disabled. Furthermore there was discussion in the past about adding individual expand / collapses buttons on the nodes themselves which would conflict with what's shown here.

Giving the user the ability to expand or collapse all and then modify from that state would make sense.

**Comment by jeff-phillips-18** [comment]
@christianvogt https://docs.google.com/document/d/1xVBWoaBDDAqzgD18SrCt4GZZiSvznSkTmzGsEpILmJM

File: frontend/packages/dev-console/src/components/topology/filters/FilterDropdown.tsx [hunk 1]
```diff
@@ -1,76 +1,101 @@
...
+  const showFilters = filters
+    .filter((f) => f.type === TopologyDisplayFilterType.show && supportedFilters.includes(f.id))
+    .sort((a, b) => a.priority - b.priority);
+
+  const selectContent = (
+    <div className="odc-topology-filter-dropdown">
+      {expandFilters.length && (
+        <div className="odc-topology-filter-dropdown__group">
+          <span
+            key="expand-groups-switch"
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/filters/FilterDropdown.tsx:49]: key isn't required here.

File: frontend/packages/dev-console/src/components/topology/filters/FilterDropdown.tsx [hunk 2]
```diff
@@ -1,76 +1,101 @@
...
+            className="odc-topology-filter-dropdown__expand-groups-switcher"
+          >
+            <span className="pf-c-select__menu-group-title">Expand</span>
+            <Switch
+              id="groups-collapse-switch"
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/filters/FilterDropdown.tsx:54]: Remove `id` attribute.

File: frontend/packages/dev-console/src/components/topology/topology-types.ts [hunk 3]
```diff
@@ -42,7 +42,11 @@ export type TopologyDisplayOption = {
 export type DisplayFilters = TopologyDisplayOption[];
 
 // Applies the filters on the model and returns the ids of filters that were relevant
-export type TopologyApplyDisplayOptions = (model: Model, filters: DisplayFilters) => string[];
+export type TopologyApplyDisplayOptions = (
+  model: Model,
+  filters: DisplayFilters,
+  groupsExpanded?: boolean,
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/topology/topology-types.ts:48]: Why isn't `groupsExpanded` just another type of display filter?
I was thinking that any display options could be handled the same way.

Then hide the usage behind a single hook which can fetch the expand display filter and only if the expand filter is `true` does it fetch the individual expand filter.
```
const isExpanded = useExpand(MY_DISPLAYFILTER);
```

**Pull request merged**